Libraries/rocThrust/device_ptr/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+# MIT License
+#
+# Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
```

Comment:
Is there a way to have the requirements shared across the examples?  That is it seems that all the examples need CMake 3.21, C++ 17, &c.

Reply:
Yes, there currently is quite some duplication between the build scripts among the examples. This is a deliberate choice and serves a dual purpose:
- Each example can be copied out-of-tree and work with minimal modifications (only the utility header has to be copied with it),
- The build scripts serve as part of the tutorial itself, as (hopefully) they are easy to follow.

We thought that in the case of the build scripts the benefits above justify the duplication. If you think otherwise, we can introduce a "build script utility library", which would reduce much of it -- in that case I think not only the C++ standard and CMake version requirement should be shared, but also a lot of the logic that is required in the build scripts. A good example for that is the [CMakeLists.txt of the rocBLAS-examples](https://github.com/ROCmSoftwarePlatform/rocBLAS-Examples/blob/develop/CMakeLists.txt) which generates all CMake projects from the tree.

Please let us know if we should make this change in build scripts "philosophy" and we'll update the PRs accordingly.

Reply:
The response makes a good point.  Leave the code as-is.